nemo_aligner/models/nlp/gpt/megatron_gpt_dpo_model.py

```diff
@@ -361,8 +363,23 @@ def get_ref_policy_logprobs(self, batch):
         pos_ids = torch.cat((batch["position_ids"], batch["position_ids"]), dim=0)
         labels = torch.cat((batch["chosen_labels"], batch["rejected_labels"]), dim=0)
         global_batch = [tokens, masks, pos_ids, labels]
-        with cpu_weight_swap(self, self.ref_policy_state_dict, megatron_amp_O2=self.megatron_amp_O2):
+
+        if self.use_peft and self.ref_policy_state_dict is None:
```

Comment:
Where is `self.use_peft` declared? There doesn't appear to be anything in the constructor of the MegatronGPTDPOModel class.

Reply:
It is defined in ```NLPAdapterModelMixin``` [class](https://github.com/NVIDIA/NeMo/blob/27d8b20ed7f95e2a93578e8ee831ed71f56e0e2a/nemo/collections/nlp/parts/mixins/nlp_adapter_mixins.py#L65) which ```MegatronGPTDPOModel``` inherits from.